packages/core/test/session.test.ts

```diff
@@ -474,6 +476,78 @@ describe('session', () => {
         expect.objectContaining({ auth })
       )
     })
+
+    it('should round up sub milliseconds transaction timeouts', async () => {
+      return await fc.assert(
+        fc.asyncProperty(
+          fc.float({ min: 0, noNaN: true }),
+          async (timeout: number) => {
+            const connection = mockBeginWithSuccess(newFakeConnection())
+
+            const { session } = setupSession({
+              connection,
+              beginTx: false,
+              database: 'neo4j'
+            })
+
+            await session.beginTransaction({ timeout })
+
+            expect(connection.seenBeginTransaction[0][0].txConfig.timeout)
+              .toEqual(int(Math.ceil(timeout)))
+          }
+        )
+      )
+    })
+
+    it('should log a warning for timeout configurations with sub milliseconds', async () => {
+      return await fc.assert(
+        fc.asyncProperty(
+          fc
+            .float({ min: 0, noNaN: true })
+            .filter((timeout: number) => !Number.isInteger(timeout)),
+          async (timeout: number) => {
+            const connection = mockBeginWithSuccess(newFakeConnection())
+
+            const { session, loggerFunction } = setupSession({
+              connection,
+              beginTx: false,
+              database: 'neo4j'
+            })
+
+            await session.beginTransaction({ timeout })
+
+            expect(loggerFunction).toBeCalledWith(
+              'info',
+              `Transaction timeout expected to be an integer, got: ${timeout}. The value will be round up.`
+            )
+          }
+        )
+      )
+    })
+
+    it('should not log a warning for timeout configurations without sub milliseconds', async () => {
+      return await fc.assert(
+        fc.asyncProperty(
+          fc.nat(),
+          async (timeout: number) => {
+            const connection = mockBeginWithSuccess(newFakeConnection())
+
+            const { session, loggerFunction } = setupSession({
+              connection,
+              beginTx: false,
+              database: 'neo4j'
+            })
+
+            await session.beginTransaction({ timeout })
+
+            expect(loggerFunction).not.toBeCalledWith(
+              'info',
+              `Transaction timeout expected to be an integer, got: ${timeout}. The value will be round up.`
+            )
```

Comment:
It seems this could be quire fragile. If you change the log message but not this test, it wouldn't start to fail, I believe.

Reply:
Yes, but for me this is fine. A change in behaviour might trigger a change in the test code. 

Of course, I will not be that specific in a testkit scenario since the change there will be expensive

Reply:
Got it, this about the `.not`. I will fix it.